packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -49,7 +55,13 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
       if (!this.signer) {
         switch (args.method) {
           case 'eth_requestAccounts': {
-            const signerType = await this.requestSignerSelection();
+            const scwOnboardMode = Array.isArray(args?.params)
+              ? (args.params[0]?.scwOnboardMode as ScwOnboardMode)
+              : undefined;
+            const options: { scwOnboardMode: ScwOnboardMode } = scwOnboardMode
+              ? { scwOnboardMode }
+              : { scwOnboardMode: 'default' };
+            const signerType = await this.requestSignerSelection(options);
```

Comment:
hmm, does this work?
```suggestion

const scwOnboardMode = args?.params?.[0]?.scwOnboardMode;
const options = { scwOnboardMode: scwOnboardMode ?? 'default' as ScwOnboardMode };
const signerType = await this.requestSignerSelection(options);

Reply:
Ah yeah this is much cleaner! For line 58 though, TS complains because `expression of type '0' can't be used to index type 'object | readonly unknown[]'.` Even though in practice eth request params are usually (always?) structured as an object inside an array, the eip-1193 provider spec say `params` is be an array or object so we gotta respect the possibility than an object could be passed as params

Line 59 is nice! Here's that change.  [5c9c0ac](https://github.com/coinbase/coinbase-wallet-sdk/pull/1383/commits/5c9c0acbcc6297952f4dd63245aa8ec2823efbeb)